### PR TITLE
Fix ignore

### PR DESCRIPTION
**Pull request by PLoic**
@quentinus95 Can you review and test on your side please ? :)

### CODE REVIEW

File: .sphinx-server.yml [hunk 1]
```diff
@@ -1,9 +1,14 @@
 # Set autobuild to true if you want to enable compilation when a file changes.
 # This will disable http authentication.
-autobuild: false
```
Comment (quentinus95) [inline .sphinx-server.yml]: autobuild should be disabled by default, please revert this change

File: server.py [hunk 2]
```diff
@@ -71,10 +71,16 @@ def pushd(new_dir):
         os.makedirs(build_folder)
 
     if configuration.get('autobuild'):
+
+        ignored_files = []
+        for ig in configuration.get('ignore'):
```
Comment (quentinus95) [inline server.py]: rename the var to `path` instead of `ig`

File: .sphinx-server.yml [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
+autobuild: true
 # Add files and folders you want autobuild watcher to ignore on changes.
 ignore:
-  - .git/
+  - .git
+
+#If you want to use a regex instead of provide name
```
Comment (quentinus95) [inline .sphinx-server.yml]: missing space after #

Comment (quentinus95) [reply .sphinx-server.yml]: Change comment to: `You can also use regex to ignore specific type of files and folders.`

File: server.py [hunk 4]
```diff
@@ -86,7 +92,8 @@ def pushd(new_dir):
         server.serve(port=8000, host='0.0.0.0', root=build_folder)
     else:
         # Building once when server starts
-        builder = sphinx_autobuild.SphinxBuilder(outdir=build_folder, args=['-b', 'html', source_folder, build_folder])
+        builder = sphinx_autobuild.SphinxBuilder(outdir=build_folder,
+                                                 args=['-b', 'html', source_folder, build_folder])
```
Comment (quentinus95) [inline server.py]: remove the line break

**Review by quentinus95 — CHANGES_REQUESTED** [review]

**Comment by quentinus95** [comment]
You need to update the Readme too.

**Comment by quentinus95** [comment]
ping @PLoic

**Comment by PLoic** [comment]
Done @quentinus95 :)

**Comment by quentinus95** [comment]
Container is crashing with the following `.sphinx-server.yml` file :

```yml

 autobuild: true
 
 ignore:
   - .git/
   - nope
   - lol/
 
 ignore_regex:
   - "*.tmp"
   - "lol*"
 
```

**Comment by PLoic** [comment]
@quentinus95 Do you have some logs ?

**Comment by quentinus95** [comment]
```
Traceback (most recent call last):
  File "/opt/sphinx-server/server.py", line 83, in <module>
    regex_ignored=configuration.get('ignore_regex')
  File "/usr/lib/python2.7/site-packages/sphinx_autobuild/__init__.py", line 137, in __init__
    self._regex_ignored = [re.compile(r) for r in regex_ignored or []]
  File "/usr/lib/python2.7/re.py", line 194, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python2.7/re.py", line 251, in _compile
    raise error, v # invalid expression
sre_constants.error: nothing to repeat
```

**Comment by quentinus95** [comment]
up !

BTW, I'm currenty using the following configuration in a project:

```yml
ignore:
  - .git/
  - .idea/
  - _build/
  - "*.pyc"
```

and it *seems* to be working, I can investigate more if you want.

**Comment by PLoic** [comment]
@quentinus95 Sorry for the time, I remove the support of the Regex for the time, but your first regex not looks like valid (http://stackoverflow.com/a/31386623)

**Review by quentinus95 — APPROVED** [review]

**Comment by quentinus95** [comment]
:+1: looks good for me

**Pull request merged**